src/main/java/com/demo/contact/service/ContactService.java

```diff
@@ -4,119 +4,149 @@
+        for (Contact contact : list) {
+            contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
+            contact.add(
+                    linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
+            contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
+                    .withRel("view"));
+        }
+        return list;
+    }
+
+    public Iterable<Contact> listContactsByPhoneNumber(String phoneNumber) {
+        Iterable<Contact> list = contactRepository
+                .listContactsByPhoneNumber("%" + phoneNumber.trim().replace(" ", "%") + "%");
+        for (Contact contact : list) {
+            contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
+            contact.add(
+                    linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
+            contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
+                    .withRel("view"));
+        }
+        return list;
+    }
+
+    public Iterable<Contact> listContactsByCity(String city) {
+        Iterable<Contact> list = contactRepository.listContactsByCity("%" + city.trim().replace(" ", "%") + "%");
+        for (Contact contact : list) {
+            contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
+            contact.add(
+                    linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
+            contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
+                    .withRel("view"));
+        }
+        return list;
+    }
+
+    public Iterable<Contact> listContactsByState(String state) {
+        Iterable<Contact> list = contactRepository.listContactsByState("%" + state.trim().replace(" ", "%") + "%");
+        for (Contact contact : list) {
+            contact.add(linkTo(ContactController.class).slash(contact.getContactId()).withSelfRel());
+            contact.add(
+                    linkTo(methodOn(ContactController.class).deleteContact(contact.getContactId())).withRel("delete"));
+            contact.add(linkTo(methodOn(ContactController.class).getProfilePicture(contact.getContactId()))
+                    .withRel("view"));
+        }
+        return list;
+    }
+
+    public MovieScore saveMovieScore(MovieScore movieScore) {
+        return movieScoreRepository.save(movieScore);
+    }
+
+    public MovieScore saveMovieScore(Long contactId, Long movieId, MovieRequest movieRequest) {
+        MovieScore movieScore = new MovieScore();
+        Contact contact = getContactById(contactId);
+        Movie movie = movieService.findMovieById(movieId);
```

Comment:
A service should not call another service. You could call the find by id from the Movie Respository instead